sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala

```diff
@@ -194,21 +198,26 @@ object ScalaJSPlugin extends AutoPlugin {
 
         scalaJSLinkerConfig := StandardConfig(),
 
+        scalaJSLinkerImplBox := new CacheBox,
+
         scalaJSLinkerImpl := {
           val s = streams.value
           val log = s.log
           val retrieveDir = s.cacheDirectory / "scalajs-linker" / scalaJSVersion
-          val lm = {
-            import sbt.librarymanagement.ivy._
-            val ivyConfig = InlineIvyConfiguration()
-              .withResolvers(Vector(Resolver.defaultLocal, Resolver.mavenCentral))
-              .withLog(log)
-            IvyDependencyResolution(ivyConfig)
+
+          scalaJSLinkerImplBox.value.ensure {
+            val lm = {
+              import sbt.librarymanagement.ivy._
+              val ivyConfig = InlineIvyConfiguration()
+                .withResolvers(Vector(Resolver.defaultLocal, Resolver.mavenCentral))
+                .withLog(log)
+              IvyDependencyResolution(ivyConfig)
+            }
+            lm.retrieve(
+                "org.scala-js" % "scalajs-linker_2.12" % scalaJSVersion,
+                scalaModuleInfo = None, retrieveDir, log)
+              .fold(w => throw w.resolveException, LinkerImpl.default _)
```

Comment:
Actually, have you considered caching this based on the classpath we pass in? That might solve the whole boxing issue.

Reply:
I don't understand how that would help. We would still need to store the cached `LinkerImpl` somewhere. Where do we cache it then? In a `Map[Seq[File], LinkerImpl]`? Where do we store that map? When do we remove entries from that map?

I guess I didn't understand what you are suggesting. Could you explain further?

Reply:
I meant something like `private var defaultLinkerImpl: Option[(Seq[File], LinkerImpl)] = None`. In *our* implementation we check against this cache.

This on its own doesn't solve the problem for clients yet. Maybe for now, we should make `LinkerImpl.default` private and expect clients that want a different linker to just depend on it directly?

Thinking of it, we should also probably do this instead of having the key `scalaJSGlobalIRCacheBox`; because it does not need to be scoped (unlike `scalaJSLinkerCacheBox`).